atlasdb-cassandra-integration-tests/src/test/java/com/palantir/atlasdb/keyvalue/cassandra/CassandraNamespaceDeleterIntegrationTest.java

```diff
@@ -0,0 +1,183 @@
+    private final Namespace namespaceTwo = Namespace.create("test_keyspace");
+    private final CassandraKeyValueServiceConfig keyValueServiceConfigForNamespaceTwo =
+            ImmutableCassandraKeyValueServiceConfig.builder()
+                    .from(CASSANDRA.getConfig())
+                    .keyspace(namespaceTwo.getName())
+                    .build();
+
+    // We must make a fresh KVS and re-initialise to get a new KVS + tables since getDefaultKvs() uses the same KVS for
+    // all tests
+    // We don't need a kvs for namespaceTwo, because we manually create the keyspace and don't care about the
+    // existence of the atlas tables for the new namespace
+    private final KeyValueService kvs = CassandraKeyValueServiceImpl.createForTesting(
+            keyValueServiceConfigWithNamespaceOne, keyValueServiceRuntimeConfig);
+    private final NamespaceDeleter namespaceDeleterForNamespaceOne =
+            new CassandraNamespaceDeleter(keyValueServiceConfigWithNamespaceOne, this::createClient);
+    private final NamespaceDeleter namespaceDeleterForNamespaceTwo =
+            new CassandraNamespaceDeleter(keyValueServiceConfigForNamespaceTwo, this::createClient);
+
+    @After
+    public void after() {
+        kvs.close();
+        namespaceDeleterForNamespaceOne.deleteAllDataFromNamespace();
+        namespaceDeleterForNamespaceTwo.deleteAllDataFromNamespace();
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDropsOnlyKeyspaceInConfig() {
+        createNamespaceTwo();
+        assertNamespaceExists(namespaceOne);
+        assertNamespaceExists(namespaceTwo);
+
+        namespaceDeleterForNamespaceOne.deleteAllDataFromNamespace();
+
+        assertNamespaceDoesNotExist(namespaceOne);
+        assertNamespaceExists(namespaceTwo);
+    }
+
+    @Test
+    public void deleteAllDataFromNamespaceDoesNotThrowIfKeyspaceDoesNotExist() {
+        assertNamespaceDoesNotExist(namespaceTwo);
+        assertThatCode(namespaceDeleterForNamespaceTwo::deleteAllDataFromNamespace)
+                .doesNotThrowAnyException();
+    }
+
+    @Test
+    public void isNamespaceDeletedSuccessfullyReturnsFalseIfKeyspaceExists() {
+        assertNamespaceExists(namespaceOne);
+
+        assertThat(namespaceDeleterForNamespaceOne.isNamespaceDeletedSuccessfully())
+                .isFalse();
+    }
+
+    @Test
+    public void isNamespaceDeletedSuccessfullyReturnsTrueIfKeyspaceDeleted() {
+        assertNamespaceExists(namespaceOne);
+
+        namespaceDeleterForNamespaceOne.deleteAllDataFromNamespace();
+
+        assertNamespaceDoesNotExist(namespaceOne);
+        assertThat(namespaceDeleterForNamespaceOne.isNamespaceDeletedSuccessfully())
+                .isTrue();
+    }
+
+    @Test
+    public void isNamespaceDeletedSuccessfullyReturnsTrueIfDifferentKeyspaceExists() {
+        // Cleanup all KVS tables and the associated keyspace
```

Comment:
I'm not sure why this use of the method warrants a comment when the others don't - what's different about it?

Reply:
Nothing really, it was just a new test I added later and forgot to add the actual cleanup steps originally. I'll just remove it, doesn't add much as you say